source/extensions/filters/http/cache/cache_header_utility.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include "common/http/header_map_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+// Byte range from an HTTP request.
+class RawByteRange {
+public:
+  // - If first==UINT64_MAX, construct a RawByteRange requesting the final last
+  // body bytes.
+  // - Otherwise, construct a RawByteRange requesting the [first,last] body
+  // bytes. Prereq: first == UINT64_MAX || first <= last Invariant: isSuffix() ||
+  // firstBytePos() <= lastBytePos
+  RawByteRange(uint64_t first, uint64_t last) : first_byte_pos_(first), last_byte_pos_(last) {
+    RELEASE_ASSERT(isSuffix() || first <= last, "Illegal byte range.");
+  }
+  bool isSuffix() const { return first_byte_pos_ == UINT64_MAX; }
+  uint64_t firstBytePos() const {
+    ASSERT(!isSuffix());
+    return first_byte_pos_;
+  }
+  uint64_t lastBytePos() const {
+    ASSERT(!isSuffix());
+    return last_byte_pos_;
+  }
+  uint64_t suffixLength() const {
+    ASSERT(isSuffix());
+    return last_byte_pos_;
+  }
+
+private:
+  uint64_t first_byte_pos_;
+  uint64_t last_byte_pos_;
+};
+
+class CacheHeaderUtility {
```

Comment:
Wrapping static functions in this class serves no purpose.

Reply:
This is mirroring the pattern seen elsewhere in Envoy: https://github.com/envoyproxy/envoy/blob/master/source/common/http/header_utility.h

I recognize it is not following Google's style guide, but I was trying to stay consistent within Envoy. If not in this class, where should this function live? Perhaps defined on HttpCache or another existing class?

Reply:
I'd put them directly in the Cache namespace. I don't know which way the maintainers will ultimately want, so go whichever way you prefer.